### ISSUE TITLE
[Feature Request] Layer Tap-Lock

### ISSUE THREAD

**Issue by kosorin** [qmk/qmk_firmware#17261]
## Feature Request Type

- [x] Core functionality
- [ ] Add-on hardware support (eg. audio, RGB, OLED screen, etc.)
- [ ] Alteration (enhancement/optimization) of existing feature(s)
- [x] New behavior

## Description

I would like a new macro for swytiching layers, combination of `MO` and `TT`. What I don't like about `TT` is that I have to always remember toggle state. 
This new feature would be more like `MO`. If I hold the key I would always be in the layer, same as `MO`. Using `TT` I can somtimes tutn off the layer with hold which is confusing for me. 
Of course tapping `TAPPING_TOGGLE` times would turn on and lock the layer.

Basically it would be implemented exactly as `TT` except these lines:

https://github.com/qmk/qmk_firmware/blob/0e11b511e4a3c48a67de6414b0907ec26dfcdf49/quantum/action.c#L592-L603

New `OP_TAP_LOCK`:

```C
case OP_TAP_LOCK:
    if (event.pressed) {
        if (tap_count <= TAPPING_TOGGLE) {
            layer_on(action.layer_tap.val);
        }
    } else {
        if (tap_count < TAPPING_TOGGLE) {
            layer_off(action.layer_tap.val);
        }
    }
    break;
```

Or just add an opnion to configure `TT` behavior.
